### PR TITLE
Create Mac-OS-X-Compile-From-Source-Instructions.md

### PR DESCRIPTION
**Pull request by maxieds**
New documentation file to provide installation from source instructions on MacOS. This procedure has been tested on MacOS Big Sur.

### CODE REVIEW

File: doc/md/Installation_Instructions/Mac-OS-X-Compile-From-Source-Instructions.md [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+vim Makefile.platform
+```
+
+## Compilation from source
+^[Top](#top)
+
+The following command has been shown to work on MacOS Big Sur with the 
+default Compiler Tools that comes installed on this release of the platform:
+```bash
+make clean && CFLAGS="-I /usr/local/opt/openssl/include" make
```
Comment (iceman1001) [inline doc/md/Installation_Instructions/Mac-OS-X-Compile-From-Source-Instructions.md:45]: I believe @doegox  pushed a fix for building on OSX;   so the CLFAGS part here shouldn't be needed.

**Comment by iceman1001** [comment]
Nice write-up,   I posted some quick inline comments.

**Comment by maxieds** [comment]
@iceman1001 
I amended the docs with your suggestions. Is it ready to merge?

**Comment by iceman1001** [comment]
Thanks!

**Comment by iceman1001** [comment]
I realise we already have this one:

https://github.com/RfidResearchGroup/proxmark3/blob/master/doc/md/Installation_Instructions/Mac-OS-X-Homebrew-Installation-Instructions.md#homebrew-mac-os-x-developer-installation

which is a the proper guide to follow when doing `homebrew` based development setup

**Pull request merged**